e2e/test/scenarios/actions/reproductions/32974-linked-parameter-via-url-does-not-work.cy.spec.js

```diff
@@ -0,0 +1,170 @@
+    native: {
+      query: `UPDATE ${TEST_TABLE} SET PRICE = 999 WHERE ID = {{ ${ID_ACTION_PARAMETER.slug} }}`,
+      "template-tags": {
+        [ID_ACTION_PARAMETER.slug]: {
+          id: ID_ACTION_PARAMETER.id,
+          "display-name": ID_ACTION_PARAMETER.name,
+          name: ID_ACTION_PARAMETER.slug,
+          type: "text",
+        },
+      },
+    },
+    database: SAMPLE_DB_ID,
+  },
+  visualization_settings: {
+    fields: {
+      [ID_ACTION_PARAMETER.id]: {
+        id: ID_ACTION_PARAMETER.id,
+        required: true,
+        fieldType: "number",
+        inputType: "number",
+      },
+    },
+  },
+};
+
+describe("Issue 32974", { tags: ["@external", "@actions"] }, () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    cy.createQuestion(MODEL_DETAILS, {
+      wrapId: true,
+      idAlias: "modelId",
+    });
+    setActionsEnabledForDB(SAMPLE_DB_ID, true);
+  });
+
+  it("can submit query action linked with dashboard parameters (metabase#32974)", () => {
+    cy.get("@modelId").then(modelId => {
+      createAction({ ...QUERY_ACTION, model_id: modelId }).then(
+        ({ body: { id: actionId } }) => {
+          cy.wrap(actionId).as("actionId");
+        },
+      );
+    });
+
+    setupDashboard();
+
+    cy.get("@dashboardId").then(dashboardId => {
+      cy.visit({ url: `/dashboard/${dashboardId}`, qs: { id: 1 } });
+    });
+
+    cy.log("Execute action");
+    cy.button(QUERY_ACTION.name).click();
+    modal().button(QUERY_ACTION.name).click();
+
+    cy.log("Assertions");
+    getDashboardCard().findByText("999").should("exist");
+    modal().should("not.exist");
+    undoToast().findByText("Query action ran successfully").should("exist");
+  });
+});
+
+function setupDashboard() {
+  cy.createDashboard(DASHBOARD_DETAILS).then(
+    ({ body: { id: dashboardId } }) => {
+      cy.wrap(dashboardId).as("dashboardId");
+    },
+  );
+
+  cy.then(function () {
+    updateDashboardCards({
+      dashboard_id: this.dashboardId,
+      cards: [
+        {
+          card_id: this.modelId,
+          // Map dashboard parameter to PRODUCTS.ID
+          parameter_mappings: [
+            {
+              parameter_id: ID_DASHBOARD_PARAMETER.id,
+              card_id: this.modelId,
+              target: ["dimension", ["field-id", PRODUCTS.ID, null]],
+            },
+          ],
+        },
+        {
+          action_id: this.actionId,
+          // Visualization settings when we add an action to a dashboard
+          visualization_settings: {
+            actionDisplayType: "button",
+            "button.label": QUERY_ACTION.name,
+            virtual_card: {
+              archived: false,
+              dataset_query: {},
+              display: "action",
+              name: null,
+              visualization_settings: {},
+            },
+          },
+          size_x: 4,
+          size_y: 1,
+          // Map action's ID parameter to dashboard parameter
+          parameter_mappings: [
```

Comment:
This is where we map an action parameter to a dashboard parameter
![image](https://github.com/metabase/metabase/assets/1937582/c6dcb0d6-aef8-489c-8565-b45c5ad71518)